Right now it does not find the glibc/stdlibc++ headers:

```
[nix-shell:~/.cache/nixpkgs-review/pr-75946]$ cat main.cpp
#include <iostream>

int main() {
  std::cout << "Hello world" << std::endl;
  return 0;
}

[nix-shell:~/.cache/nixpkgs-review/pr-75946]$ cat main.c
#include <stdio.h>

int main() {
  printf("Hello world\n");
  return 0;
}
[nix-shell:~/.cache/nixpkgs-review/pr-75946]$ compute++ main.cpp -o main
/nix/store/ymhg9r0xambgxpzj0gh6l6lcnk69ykdq-computecpp-1.2.0/bin/compute++: /nix/store/cbl3mpw1hr0b6038j5m9mp9hi38iyjjx-ncurses-6.1-20190112-abi5-compat/lib/libtinfo.so.5: no version information available (required by /nix/store/ymhg9r0xambgxpzj0gh6l6lcnk69ykdq-computecpp-1.2.0/bin/compute++)
/nix/store/ymhg9r0xambgxpzj0gh6l6lcnk69ykdq-computecpp-1.2.0/bin/compute++: /nix/store/cbl3mpw1hr0b6038j5m9mp9hi38iyjjx-ncurses-6.1-20190112-abi5-compat/lib/libtinfo.so.5: no version information available (required by /nix/store/ymhg9r0xambgxpzj0gh6l6lcnk69ykdq-computecpp-1.2.0/bin/compute++)
main.cpp:1:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.

[nix-shell:~/.cache/nixpkgs-review/pr-75946]$ compute main.c -o main
/nix/store/ymhg9r0xambgxpzj0gh6l6lcnk69ykdq-computecpp-1.2.0/bin/compute: /nix/store/cbl3mpw1hr0b6038j5m9mp9hi38iyjjx-ncurses-6.1-20190112-abi5-compat/lib/libtinfo.so.5: no version information available (required by /nix/store/ymhg9r0xambgxpzj0gh6l6lcnk69ykdq-computecpp-1.2.0/bin/compute)
/nix/store/ymhg9r0xambgxpzj0gh6l6lcnk69ykdq-computecpp-1.2.0/bin/compute: /nix/store/cbl3mpw1hr0b6038j5m9mp9hi38iyjjx-ncurses-6.1-20190112-abi5-compat/lib/libtinfo.so.5: no version information available (required by /nix/store/ymhg9r0xambgxpzj0gh6l6lcnk69ykdq-computecpp-1.2.0/bin/compute)
In file included from main.c:1:
/nix/store/4v4ginzfvpsrnvafl1cxx1jn4wy6kc1r-glibc-2.27-dev/include/stdio.h:33:10: fatal error: 'stddef.h' file not found
#include <stddef.h>
         ^~~~~~~~~~
1 error generated.
```

--- a/pkgs/development/compilers/computecpp/default.nix
+++ b/pkgs/development/compilers/computecpp/default.nix
@@ -1,0 +1,52 @@
+{ stdenv
+, fetchzip
+, pkg-config
+, autoPatchelfHook
+, installShellFiles
+, ncurses5
+, ocl-icd
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "computecpp";
+  version = "1.2.0";
+
+  src = fetchzip {
+    url = "https://computecpp.codeplay.com/downloads/computecpp-ce/${version}/ubuntu-16.04-64bit.tar.gz";
+    sha256 = "191kwvzxfg1sbaq6aw6f84chi7bhsibb2a63zsyz3gz8m0c0syr5";
+    stripRoot = true;
+  };
+
+  dontStrip = true;
+
+  buildInputs = [ stdenv.cc.cc.lib ncurses5 ocl-icd zlib ];
+  nativeBuildInputs = [ autoPatchelfHook pkg-config installShellFiles ];
+
+  installPhase = ''
+    runHook preInstall
+
+    find ./lib -type f -exec install -D -m 0755 {} -t $out/lib \;
+    find ./bin -type l -exec install -D -m 0755 {} -t $out/bin \;
+    find ./bin -type f -exec install -D -m 0755 {} -t $out/bin \;
+    find ./doc -type f -exec install -D -m 0644 {} -t $out/doc \;
+    find ./include -type f -exec install -D -m 0644 {} -t $out/include \;
+
+    runHook postInstall
+  '';
+
+  passthru = {
+    isClang = true;
+  } // stdenv.lib.optionalAttrs (stdenv.targetPlatform.isLinux || (stdenv.cc.isGNU && stdenv.cc.cc ? gcc)) {
+    gcc = if stdenv.cc.isGNU then stdenv.cc.cc else stdenv.cc.cc.gcc;
+  };
+
+  meta = with stdenv.lib; {
+    description =
+      "Accelerate Complex C++ Applications on Heterogeneous Compute Systems using Open Standards";
+    homepage = https://www.codeplay.com/products/computesuite/computecpp;
+    license = licenses.unfree;
+    maintainers = with maintainers; [ davidtwco ];
+    platforms = [ "x86_64-linux" ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7742,6 +7742,28 @@ in
 
   compcert = callPackage ../development/compilers/compcert { };
 
+  computecpp-unwrapped = callPackage ../development/compilers/computecpp {};
+  computecpp = wrapCCWith rec {
+    cc = computecpp-unwrapped;
+    extraPackages = [
+      libstdcxxHook
+      llvmPackages.compiler-rt
+    ];
+    extraBuildCommands = ''
+      wrap compute $wrapper $ccPath/compute
+      wrap compute++ $wrapper $ccPath/compute++
+      export named_cc=compute
+      export named_cxx=compute++
+
+      rsrc="$out/resource-root"
+      mkdir -p "$rsrc/lib"
+      ln -s "${cc}/lib" "$rsrc/include"
+      echo "-resource-dir=$rsrc" >> $out/nix-support/cc-cflags
+    '' + stdenv.lib.optionalString (stdenv.targetPlatform.isLinux && cc ? gcc && !(stdenv.targetPlatform.useLLVM or false)) ''
+      echo "--gcc-toolchain=${cc.gcc}" >> $out/nix-support/cc-cflags
+    '';
+  };
+
   cryptol = haskell.lib.justStaticExecutables haskellPackages.cryptol;
 
   inherit (callPackages ../development/compilers/crystal {

